src/nvim/ops.c

```diff
@@ -1028,7 +1028,8 @@ static void put_reedit_in_typebuf(int silent)
       buf[1] = 'R';
       buf[2] = NUL;
     } else {
-      buf[0] = restart_edit == 'I' ? 'i' : restart_edit;
+      assert(restart_edit <= UCHAR_MAX);
```

Comment:
not needed as restart_edit is only ever set to 0 or character constant. It is probably better (in a later PR) to change its type to char or char_u.